test/test_pynisher.py

```diff
@@ -322,45 +340,51 @@ def test_liblinear_svc(self):
         time.sleep(1)
         p = psutil.Process()
         self.assertEqual(len(p.children(recursive=True)), expected_children[context])
-        self.assertEqual(logger_mock.debug.call_count, 2)
-        self.assertEqual(logger_mock.debug.call_args_list[0][0][0],
+        # Using a picklable-logger to capture all messages
+        self.assertEqual(this_logger.debug.call_count, 4)
+        self.assertEqual(this_logger.debug.call_args_list[0][0][0],
+                         'Restricting your function to 2 seconds cpu time.')
+        self.assertEqual(this_logger.debug.call_args_list[1][0][0],
+                         'Allowing a grace period of 1 seconds.')
+        self.assertEqual(this_logger.debug.call_args_list[2][0][0],
                          'Function called with argument: (16384, 10000), {}')
-        self.assertEqual(logger_mock.debug.call_args_list[1][0][0],
+        self.assertEqual(this_logger.debug.call_args_list[3][0][0],
                          'Your function call closed the pipe prematurely -> '
                          'Subprocess probably got an uncatchable signal.')
         # self.assertEqual(wrapped_function.exit_status, pynisher.CpuTimeoutException)
-        self.assertGreater(duration, time_limit - 0.1)
-        self.assertLess(duration, time_limit + grace_period + 0.1)
-        self.assertEqual(wrapped_function.exitcode, -9)
+        if context == 'fork':
+            self.assertGreater(duration, time_limit - 0.1)
+            self.assertLess(duration, time_limit + grace_period + 0.1)
+            self.assertEqual(wrapped_function.exitcode, -9)
 
     @unittest.skipIf(not all_tests, "skipping nested pynisher test")
     def test_nesting(self):
 
-        tl = 2  # time limit
-        gp = 1  # grace period
+        tl = 3  # time limit
+        gp = 2  # grace period
```

Comment:
The grace period appears to depend on the context, and as spawn and forkserver have a higher overhead, a lower grace period can result test-unrelated failures. Could you please make those times dependent on the context?

Reply:
Done!